### PR TITLE
Liche dive removed, model simplified (one bomb), cannot fire from landed

### PR DESCRIPTION
**Pull request by StanczakDominik**
### Work done

Hornet:
- fixed the Liche unit cannon exploit by removing VTOL modifier
- removed liche dive by doing some magic I don't understand myself, explain more plz
- removed the turret mode where liches could hilariously fire while landed
- fixed model to not have two bombs and rotate between them, iirc?

#### Test steps
- [x] Liche should no longer fire when landed, it should now fly up first
- [x] Liche should not be able to catapult t1 bombers etc by firing near them
- [x] Liche should no longer dive - double check set target/attack behavior is comfortable
- [x] look under the liche while it fires and check what it does with the bombs

### CODE REVIEW

**Comment by StanczakDominik** [comment]
# Liche should no longer fire when landed, it should now fly up first
I think it's still able to shoot from landed *right after it lands or it gets spawned*, which is fine tbh. Most of the problematic behavior is gone.
# Liche should not be able to catapult t1 bombers etc by firing near them
Well they get catapulted but they're also dead so it's ok :D
# Liche should no longer dive - double check set target/attack behavior is comfortable
They do not! I'll still test on a tall map to be sure
# look under the liche while it fires and check what it does with the bombs
screen_2024-03-02_19-05-05-245
screen_2024-03-02_19-05-38-993
correct bomb behavior

**Comment by StanczakDominik** [comment]
I tested on Industrial Revolution spawned from one of the tall towers, but it doesn't seem to be a specifically attack-caused dive, more a "weird air unit behavior near tall towers". I think we can consider it fixed.
screen_2024-03-02_19-15-44-293


Most importantly, set-target behavior seems retained as far as I can tell, which lets you still do the fancy micro while not making this suck for new players :)

File: luarules/gadgets/unit_liche_cruisecontrol.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+
+
+local licheId = UnitDefNames.armliche.id
```
Comment (thehobojoe) [inline luarules/gadgets/unit_liche_cruisecontrol.lua:22]: As with the infestor code, even though `armliche` is unlikely to be undefined it's best to be certain.
```
local licheId 
local liche = UnitDefNames.armliche
if liche then 
    licheId = liche .id 
else 
    gadgetHandler:RemoveGadget(self) 
end
```

File: luarules/gadgets/unit_liche_cruisecontrol.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+function gadget:UnitFinished(unitID, unitDefID, unitTeam)
+    if unitDefID == licheId then
+        local opts = {}
+        opts.attackSafetyDistance = 3000
```
Comment (thehobojoe) [inline luarules/gadgets/unit_liche_cruisecontrol.lua:27]: does this not work in the unitdefs?

File: luarules/gadgets/unit_liche_cruisecontrol.lua (hunk 1 above)
Comment (TheSilverHornet) [reply luarules/gadgets/unit_liche_cruisecontrol.lua:22]: I'd solved this in both but the code isn't here yet (perfi being slightly busy in the grand melee yesterday); the removegadget is a nice idea, will integrate

File: luarules/gadgets/unit_liche_cruisecontrol.lua (hunk 2 above)
Comment (TheSilverHornet) [reply luarules/gadgets/unit_liche_cruisecontrol.lua:27]: alas it does not. it was intentionally moved out at some point long ago, for reasons I don't understand

**Pull request merged**